script.go

```diff
@@ -928,6 +930,13 @@ func (p *Pipe) WithStdout(w io.Writer) *Pipe {
 	return p
 }
 
+// WithContext sets context.Context for the pipe. Adds support for graceful pipe
+// shutdown. Currently works with [Pipe.Exec] and [Pipe.ExecForEach]
+func (p *Pipe) WithContext(ctx context.Context) *Pipe {
+	p.ctx = ctx
+	return p
+}
```

Comment:
I like the new syntax, but I would have prefered if the pipe was immutable

```suggestion
func (p Pipe) WithContext(ctx context.Context) *Pipe {
	p2 := p
	p2.ctx = ctx
	return p2
}
```

Or this simpler syntax (but some linters may report a bad pattern, as we are modifying a non-pointer receiver

```suggestion
func (p Pipe) WithContext(ctx context.Context) *Pipe {
	p.ctx = ctx
	return p
}
```

But anyway, I'm fine with the feature

Reply:
I'm just noticing, all other methods are like that. Let's keep it unchanged.

It's not error prone, but at lest it's memory efficient